core/src/main/java/org/wildfly/extras/creaper/core/online/OnlineOptions.java

```diff
@@ -393,8 +395,14 @@ public void handle(Callback[] callbacks) throws IOException, UnsupportedCallback
 
         ModelControllerClient modelControllerClient;
 
-        if (protocol == ManagementProtocol.HTTP) {
-            modelControllerClient = new HttpModelControllerClient(host, port, username, password, connectionTimeout);
+        if (protocol == ManagementProtocol.HTTP || protocol == ManagementProtocol.HTTPS) {
+            if (protocol == ManagementProtocol.HTTP) {
+                modelControllerClient = HttpModelControllerClient.http(
+                        host, port, username, password, connectionTimeout);
+            } else {
+                modelControllerClient =  HttpModelControllerClient.https(
+                        host, port, username, password, connectionTimeout, sslOptions);
+            }
```

Comment:
How about just passing `(SslOptions) null` to the `HttpModelControllerClient` constructor would mean "pure HTTP", otherwise "HTTPS"?